gateway-messages/src/mgs_to_sp.rs

```diff
@@ -407,3 +409,23 @@ bitflags::bitflags! {
         const STARTUP_VERBOSE = 1 << 8;
     }
 }
+
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, SerializedSize,
+)]
+pub enum DumpRequest {
+    /// Returns the number of task crash dumps
+    TaskDumpCount,
+
+    /// Begins reading a task crash dump by index
+    ///
+    /// `key` is used for disambiguation between multiple readers
+    TaskDumpReadStart {
+        index: u32,
+        key: u32,
```

Comment:
This is probably fine, but for update uniqueness we use a UUID (converted into a `[u8;16]` at this level). I'm lightly inclined to repeat that here, for the consistency + ease of `Uuid::new_v4()` over importing rand?